controllers/cloud.redhat.com/namespaces.go

```diff
@@ -285,10 +280,40 @@ func DeletePrometheusOperator(ctx context.Context, cl client.Client, nsName stri
 	prometheusOperator.SetGroupVersionKind(gvk)
 	prometheusOperator.SetName(GetPrometheusOperatorName(nsName))
 
+	err := cl.Get(ctx, types.NamespacedName{Name: GetPrometheusOperatorName(nsName)}, &prometheusOperator)
+	if k8serr.IsNotFound(err) {
+		return true, nil
+	} else if err != nil {
+		return false, err
+	}
+
 	err = cl.Delete(ctx, &prometheusOperator)
 	if err != nil {
-		return fmt.Errorf("error deleting prometheus operator %s: %v", GetPrometheusOperatorName(nsName), err)
+		return false, fmt.Errorf("error deleting prometheus operator %s: %v", GetPrometheusOperatorName(nsName), err)
 	}
 
-	return nil
+	return true, nil
+}
+
+func CheckForSubscriptionPrometheusOperator(ctx context.Context, cl client.Client, nsName string) (bool, error) {
+	subscriptionsPrometheusOperator := unstructured.Unstructured{}
+
+	gvk := schema.GroupVersionKind{
+		Group:   "operators.coreos.com",
+		Version: "v1alpha1",
+		Kind:    "Subscription",
+	}
+
+	subscriptionsPrometheusOperator.SetGroupVersionKind(gvk)
+	subscriptionsPrometheusOperator.SetName("prometheus")
+	subscriptionsPrometheusOperator.SetNamespace(nsName)
+
+	err := cl.Get(ctx, types.NamespacedName{Name: "prometheus", Namespace: nsName}, &subscriptionsPrometheusOperator)
+	if k8serr.IsNotFound(err) {
+		return true, nil
+	} else if err == nil {
+		return false, err
+	}
+
+	return true, nil
```

Comment:
Sorry one more thing I spotted -- if we hit an err (that is not a `NotFound` error) -- we will return `true, nil`

I think we should return `false, err` in that case, right? Logic could be simplified to:

```go
	if k8serr.IsNotFound(err) {
		return true, nil
	}
        return false, err
}
```